### DISCUSSION TITLE
GitHub Token generation & scopes/permissions

### DISCUSSION THREAD

**Discussion by OtheJoyof** [cafawo/FinancialDataAnalytics#1] [Q&A] [answered]
**To clone your Github repository, you need to make your Personal Access Token**

When you are doing `git clone http://github.com/YourUserName/YourStudentID.git` 
after you put your Github username and password you will see this:

<img width="710" alt="image" src="https://user-images.githubusercontent.com/130544491/232090153-13478b28-7b2e-4703-8b56-4eae271e3dc4.png">

So I went to make a token, but if you see the screen, there are two options:
1. Fine-grained tokens
2. Tokens (classic)

![image](https://user-images.githubusercontent.com/130544491/232093232-333ee309-f830-4b71-b731-8cf0c6cb9eea.png)

Also, you can see many permissions or scopes that you would select or not.

![image](https://user-images.githubusercontent.com/130544491/232092219-ca18c616-1af5-4b59-bc56-cd5d962ba1b4.png)

I don't know which permission or scope I should select so that it will be helpful for this class. I expect I should check 'repo' scope, but for others, I'm not too sure. If someone already finished this task, please give me a good advice!

**Comment by OtheJoyof** [comment]
**+ I selected all scopes, at the end.**
I am really curious, is everybody all used to use Git??
It seems like I'm the only one wandering about and adapting in how to use this tool...😇

P.S.

While you setup tech, there are more things you need to do than just `git pull`
I faced a lot of errors and had to set my name, email, branch(main) and so on.
Even if you google to solve your problem the result not be exactly suit for your circumstances.

For example, some bloggers says to set branch as "master" but now the branch we have originally is not "master" but "main". So when you set your origin branch, you might need to use the word "main". However, I mainly searched Korean users' blogs so it may be different and less updated from English or German google results.

**Reply by OlgaKuzminichna** [reply]
I'm not entirely sure, but I always avoid using Git Bash directly. Instead, I prefer to use the built-in Git console in either VSCode or PyCharm. This way, I don't have to write code within the Git console and avoid any frustrating errors. I just attached these two repositories together and use the commit and push/pull options located in the left corner. 
@OtheJoyof Please let me now if I am missing something not using Git
![image](https://user-images.githubusercontent.com/120052171/232708847-5b4ff853-27c2-4617-8d00-e0b3fcddffaf.png)

**Comment by cafawo** [comment, marked as answer]
To answer the original question: **repo** scope is sufficient. More information at: 
* https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps
* https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token

Feel free to use any code editor you like. I use Spyder and VS Code, whereas the latter has the neat Git(/Hub) integration that @olyafrontend mentioned in her comment.

**Comment by cafawo** [comment]
Regarding the "main" / "master" discussion, see: https://github.com/github/renaming

(PS: feel free to open a new discussion for each question, which makes the Q&A more useful in the future.)
